src/server/engine_shard_set.cc

```diff
@@ -603,24 +603,55 @@ void EngineShard::Heartbeat() {
 
   CacheStats();
 
-  if (IsReplica())  // Never run expiration on replica.
-    return;
+  if (!IsReplica()) {  // Never run expiration on replica.
+    constexpr double kTtlDeleteLimit = 200;
+    constexpr double kRedLimitFactor = 0.1;
+
+    uint32_t traversed = GetMovingSum6(TTL_TRAVERSE);
+    uint32_t deleted = GetMovingSum6(TTL_DELETE);
+    unsigned ttl_delete_target = 5;
+
+    if (deleted > 10) {
+      // deleted should be <= traversed.
+      // hence we map our delete/traversed ratio into a range [0, kTtlDeleteLimit).
+      // The higher t
```

Comment:
Please continue the sentence, the suspension is killing me

Reply:
I would but for s